jupyter_server/auth/login.py

```diff
@@ -172,6 +172,8 @@ def get_user(cls, handler):
         if user_id is None:
             get_secure_cookie_kwargs  = handler.settings.get('get_secure_cookie_kwargs', {})
             user_id = handler.get_secure_cookie(handler.cookie_name, **get_secure_cookie_kwargs )
+            if user_id:
+                user_id = user_id.decode()
```

Comment:
#### Question

Should this branch set `handler._token_authenticated = True`.

Technically it is "cookie authenticated", however, the cookie is based on an earlier token.

I ask because it can be useful for a handler to know when token authorisation is in use, for example:

```python
class MyExtensionHandler(ExtensionHandlerMixin, JupyterHandler):  

    @tornado.web.authenticated
    def get(self):
        user = self.get_current_user()                                           
        if self.token_authenticated:
            user = getpass.getuser()
        self.write(f'hello {user}') 
```

Reply:
Opened #566 to follow up on this later.